sql-stream-to-stream-join-test/src/main/java/com/hazelcast/jet/tests/sql/SqlStreamToStreamJoinSoakTest.java

```diff
@@ -0,0 +1,221 @@
+        sqlService = client.getSql();
+        ingestionExecutorService = Executors.newSingleThreadExecutor();
+        startTime = System.currentTimeMillis();
+        queryTimeout = propertyInt("queryTimeout", DEFAULT_QUERY_TIMEOUT_MILLIS);
+        String brokerUri = property("brokerUri", "localhost:9092");
+
+        logger.info("Creating mapping to Kafka with brokerUri: " + brokerUri);
+        SqlResult sourceMappingCreateResult = sqlService.execute("CREATE MAPPING " + sourceName + " ("
+                + "event_time TIMESTAMP WITH TIME ZONE,"
+                + "event_tick BIGINT )"
+                + "TYPE " + KafkaSqlConnector.TYPE_NAME + ' '
+                + "OPTIONS ( "
+                + '\'' + SqlConnector.OPTION_KEY_FORMAT + "'='int'"
+                + ", '" + SqlConnector.OPTION_VALUE_FORMAT + "'='json-flat'"
+                + ", 'bootstrap.servers'='" + brokerUri + '\''
+                + ", 'auto.offset.reset'='earliest'"
+                + ")"
+        );
+        AbstractSoakTest.assertEquals(0L, sourceMappingCreateResult.updateCount());
+
+        String initialIngestionQuery = "INSERT INTO " + sourceName + " VALUES" +
+                TestRecordProducer.produceTradeRecords(EVENTS_START_TIME, EVENTS_COUNT_PER_BATCH, EVENT_TIME_INTERVAL);
+        logger.info("Initial ingestion query: " + initialIngestionQuery);
+        SqlResult initialDataIngestionResult = sqlService.execute(initialIngestionQuery);
+        AbstractSoakTest.assertEquals(0L, initialDataIngestionResult.updateCount());
+
+        sqlService.execute("CREATE VIEW v1 AS "
+                + "SELECT * FROM TABLE(IMPOSE_ORDER(TABLE "
+                + sourceName
+                + ", DESCRIPTOR(event_time), INTERVAL '1' SECOND))");
+
+        sqlService.execute("CREATE VIEW v2 AS "
+                + "SELECT * FROM TABLE(IMPOSE_ORDER(TABLE "
+                + sourceName
+                + ", DESCRIPTOR(event_time), INTERVAL '1' SECOND))");
+    }
+
+    @Override
+    protected final void test(HazelcastInstance client, String name) {
+
+        DataIngestionTask producerTask = new DataIngestionTask(sqlService, sourceName, queryTimeout);
+        ingestionExecutorService.execute(producerTask);
+
+        Util.sleepMillis(30_000L);
+
+        SqlResult sqlResult = sqlService.execute("SELECT * FROM v1 JOIN v2 ON v1.event_time=v2.event_time");
+
+        Iterator<SqlRow> iterator = sqlResult.iterator();
+        while (System.currentTimeMillis() - startTime < durationInMillis) {
+            SqlRow sqlRow = iterator.next();
+
+            // l_event       r_event
+            // time(0) | 0 | time(0) | 0  <-- joint row
+
+            // checking only event_tick to simplify our destiny.
+            Long leftRowIndex = sqlRow.getObject(1);
+            Long rightRowIndex = sqlRow.getObject(3);
+            AbstractSoakTest.assertEquals(leftRowIndex, rightRowIndex);
+
+            currentQueryCount++;
+            printProgress();
```

Comment:
Maybe we can add here some really short sleep to not stress it, WDYT?

Reply:
I'm not sure what's being stressed here, but I could be missing something. The query is only submitted once and then the streaming results are fetched with the call to `iterator.next()` if none are available. It will block until the next result(s) are received. The `DataIngestionTask` is driving this process, and it does have a `Util.sleepMillis(queryTimeout)`.

---
This does make me realize that the `SqlResult` in the above method isn't being closed. I guess it won't cause any problems in the happy case since it needs to be open until the soak test is finished anyway. However, I think it would be a leak if this test fails for some reason but others are still running.

Reply:
Makes sense, my fault. About closing `SqlResult` - it should be easy to implement (just try-with-resources) or am I miss something else?

Reply:
Yep, that's right.